common/services/event.js

```diff
@@ -0,0 +1,85 @@
+const restClient = require('../lib/api-client/rest-client')
+
+const BaseService = require('./base')
+class EventService extends BaseService {
+  postEvents(lockoutEvents, move, user) {
+    const PER_EVENTS = [
+      'PerViolentDangerous',
+      'PerWeapons',
+      'PerConcealed',
+      'PerSelfHarm',
+      'PerEscape',
+      'PerMedicalAid',
+      'PerMedicalDrugsAlcohol',
+      'PerMedicalMentalHealth',
+      'PerPropertyChange',
+      'PerGeneric',
+    ]
+    const PERSON_MOVE_EVENTS = [
+      'PersonMoveUsedForce',
+      'PersonMoveDeathInCustody',
+    ]
+
+    const todaysDate = new Date()
+    const events = []
+
+    events.push(lockoutEvents.events)
+
+    events.flat().map(async event => {
+      let type
+      let relationshipsId
+      let supplierPersonnelNumber
+
+      const eventDescription = lockoutEvents[event]
+
+      if (PER_EVENTS.includes(event)) {
+        type = 'person_escort_records'
+        relationshipsId = move.profile.person_escort_record.id
+        supplierPersonnelNumber = 'supplier_personnel_number'
+      } else if (PERSON_MOVE_EVENTS.includes(event)) {
+        type = 'moves'
+        relationshipsId = move.id
+        supplierPersonnelNumber = 'supplier_personnel_numbers'
+      }
+
+      const payload = {
+        data: {
+          type: 'events',
+          attributes: {
+            occurred_at: todaysDate.toISOString(),
+            recorded_at: todaysDate.toISOString(),
+            notes: eventDescription,
+            details: {
+              [supplierPersonnelNumber]: [user.username],
+              fault_classification: 'investigation',
+              reported_at: todaysDate.toISOString(),
+              advised_at: todaysDate.toISOString(),
+              advised_by: user.username,
+            },
+            event_type: event,
+          },
+          relationships: {
+            eventable: {
+              data: {
+                type: type,
+                id: relationshipsId,
+              },
+            },
+            location: {
+              data: {
+                id: `${move.from_location.id}`,
```

Comment:
This should actually be the location of where the event occurred, which in this case would be where the person is held over night.

Reply:
Could we use the `move.to_location.id` in this case then? as the [incident events git repo](https://github.com/ministryofjustice/hmpps-book-secure-move-api/wiki/Incident-Events) says: 

`location - The location the incident occurred or the destination the vehicle was headed`

So we could use the original `to_location` as that would be the destination the vehicle was originally headed too?

Reply:
I think ideally we'd use the police lockout location, but if that's impossible then `move.to_location.id` is better, however it wouldn't be entirely accurate (users might be confused if they see the event happened at the start location when actually it happened during a lockout).